core/connection/manager.go

```diff
@@ -408,20 +452,20 @@ func (m *connectionManager) cleanAfterDisconnect() {
 	m.cleanupAfterDisconnect = nil
 }
 
-func (m *connectionManager) createP2PChannel(ctx context.Context, consumerID, providerID identity.Identity, proposal market.ServiceProposal, tracer *trace.Tracer) error {
+func (m *connectionManager) createP2PChannel(opts ConnectOptions, tracer *trace.Tracer) error {
 	trace := tracer.StartStage("Consumer P2P channel creation")
 	defer tracer.EndStage(trace)
 
-	contactDef, err := p2p.ParseContact(proposal.ProviderContacts)
+	contactDef, err := p2p.ParseContact(opts.Proposal.ProviderContacts)
 	if err != nil {
 		return fmt.Errorf("provider does not support p2p communication: %w", err)
 	}
 
-	timeoutCtx, cancel := context.WithTimeout(ctx, p2pDialTimeout)
+	timeoutCtx, cancel := context.WithTimeout(m.currentCtx(), p2pDialTimeout)
```

Comment:
This is a bit uncommon (contexts are usually explicitly passed around). What led you to this decision?

Reply:
I just moved it from one place to another, we were still passing `m.currentCtx()`.